internal/pkg/docker/dockerfile/dockerfile.go

```
@@ -89,31 +114,62 @@ func (df *Dockerfile) parse() error {
 	file, err := df.fs.Open(df.path)
 
 	if err != nil {
-		return fmt.Errorf("read dockerfile: %w", err)
+		return fmt.Errorf("read Dockerfile: %w", err)
 	}
 	defer file.Close()
 
-	scanner := bufio.NewScanner(file)
-	parsedDockerfile := parseFromScanner(scanner)
+	f, err := afero.ReadFile(df.fs, file.Name())
+	if err != nil {
+		return fmt.Errorf("read Dockerfile %s error: %w", f, err)
+	}
+	fileString := string(f)
+
+	parsedDockerfile, err := parseFromReader(fileString)
```

```suggestion
	parsedDockerfile, err := parseFromReader(string(content))
```
This is minor, but usually unless a variable is used more than once consider inlining the temporary variable: https://refactoring.com/catalog/inlineVariable.html

It helps readability when you can get rid of a bunch of temps.

---

That is a neat way to convert it!